test/library/draft/Vector/Vector.chpl

```diff
@@ -0,0 +1,1263 @@
+
+        _leave();
+      }
+    }
+
+    /*
+      Return a zero-based index into this vector of the first item whose value
+      is equal to `x`. If no such element can be found this method returns
+      the value `-1`.
+
+      .. warning::
+
+        Calling this method on an empty vector or with values of `start` or 
+        `end` that are out of bounds will cause the currently running program
+        to halt. If the `--fast` flag is used, no safety checks will be
+        performed.
+
+      :arg x: An element to search for.
+      :type x: `eltType`
+
+      :arg start: The start index to start searching from.
+      :type start: `int`
+
+      :arg end: The end index to stop searching at. A value less than
+                `0` will search the entire vector.
+      :type end: `int`
+
+      :return: The index of the element to search for, or `-1` on error.
+      :rtype: `int`
+    */
+    proc const indexOf(x: eltType, start: int=0, end: int=-1): int {
+      if boundsChecking {
+        const msg = " index for \"vector.indexOf\" out of bounds: ";
+
+        if end >= 0 && !_withinBounds(end) then
+          boundsCheckHalt("End" + msg + end:string);
+
+        if !_withinBounds(start) then
+          boundsCheckHalt("Start" + msg + start:string);
+      }
+
+      param error = -1;
+
+      if end >= 0 && end < start then
+        return error;
+
+      var result = error;
+
+      on this {
+        _enter();
+
+        const stop = if end < 0 then _size-1 else end;
+
+        for i in start..stop do
+          if x == _data[i] {
+            result = i;
+            break;
+          }
+
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Return the number of times a given element is found in this vector.
+
+      :arg x: An element to count.
+      :type x: `eltType`
+
+      :return: The number of times a given element is found in this vector.
+      :rtype: `int`
+    */
+    proc const count(x: eltType): int {
+      var result = 0;
+
+      on this {
+        _enter();
+
+        var count = 0;
+
+        for item in this do
+          if x == item then
+            count += 1;
+
+        result = count;
+
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Sort the items of this vector in place using a comparator. If no comparator
+      is provided, sort this vector using the default sort order of its elements.
+
+      .. warning::
+
+        Sorting the elements of this vector may invalidate existing references
+        to the elements contained in this vector.
+
+      :arg comparator: A comparator used to sort this vector.
+    */
+    proc ref sort(comparator: ?rec=Sort.defaultComparator) {
+      on this {
+        _enter();
+
+        //
+        // TODO: This is not ideal, but the Sort API needs to be adjusted
+        // before we can sort over vectors directly.
+        //
+        if _size > 1 {
+
+          // Copy current vector contents into an array.
```

Comment:
I am not very sure of using extra space for sorting. Is there any other way to do this?

Reply:
You're right. https://github.com/chapel-lang/chapel/pull/16048/commits/48f88592512a2586f23e8a9deae434a9de782c70
https://github.com/chapel-lang/chapel/pull/16048/commits/8f8618b1dbc6559428f010504f97c352807e0a7a
https://github.com/chapel-lang/chapel/pull/16048/commits/b7fd0f7492cc18362339cc004b38348adbbb92ce